include/mbedtls/ecdsa.h

```diff
@@ -53,13 +53,13 @@
  *
  * For each of r and s, the value (V) may include an extra initial "0" bit.
  */
-#define MBEDTLS_ECDSA_MAX_SIG_LEN( bits )                               \
-    ( /*T,L of SEQUENCE*/ ( ( bits ) >= 61 * 8 ? 3 : 2 ) +              \
-      /*T,L of r,s*/        2 * ( ( ( bits ) >= 127 * 8 ? 3 : 2 ) +     \
-      /*V of r,s*/                ( ( bits ) + 8 ) / 8 ) )
+#define MBEDTLS_ECDSA_MAX_SIG_LEN(bits)               \
+    (/*T,L of SEQUENCE*/ ((bits) >= 61 * 8 ? 3 : 2) + \
+     /*T,L of r,s*/ 2 *                               \
```

Comment:
I guess we need to reformat the original somehow to make the reformatted version readable.

Reply:
There will be many such cases. Seems like my dream about using the clang-format as magic tool for "reformat and forget" is gone.
We will have to correct the code after clang-formating

Reply:
The important thing here is that when we are done with the reformatting, either some manual change then automatic reformatting or automatic reformatting then manual fixes, and we re-apply the automatic reformatting, nothing changes.

Reply:
You are right, please see my comment https://github.com/ARMmbed/mbedtls/pull/4794#issuecomment-884835781

Reply:
We want this to be usable by contributors on their own machine, not just by CI. It's important that a wide range of different versions of the tool give the same result.

Reply:
I would rather pick one version and stick to it rather then attempt to support a variety of versions, for following reasons:

1. Not every option of clang-format is backwards compatible. It's also possible that consecutive versions of clang-format will yield different formatting from the same code.
2. Supporiting a wide range of versions would require using the `.clang-format` file which is using only options from the lowest version of supported range. This however gives no guarantee that code formatting will be consistent across various versions, because clang-format will use default values for undefined options. If we have `.clang-format` compatible with version 4 and format the code using clang-format-4 and clang-format-6, the results might differ solely due appearance of new style options in newer version. (Versions mentioned are just examples, I haven't checked specific differences between these two versions).

Reply:
As agreed on video-call - we are going with version 12 only.